### PR TITLE
move evm_circuit::witness as crate::witness; split it into smaller mods

### PR DESCRIPTION
**Pull request by lispc**
part1 of https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/708

### CODE REVIEW

**Review by ChihChengLiang — APPROVED** [review]
LGTM

File: zkevm-circuits/src/witness.rs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#![allow(missing_docs)]
```
Comment (CPerezz) [inline zkevm-circuits/src/witness.rs:1]: Can't you simply add the minimal documentation needed?

**Review by CPerezz — APPROVED** [review]
LGTM. Just some concern about the docs. I would prefer they;'re added instead of alowing avoiding it on clippy.

The rest looks good.

File: zkevm-circuits/src/witness.rs (hunk 1 above)
Comment (lispc) [reply zkevm-circuits/src/witness.rs]: add basic docs except Rw struct which needs a lot of work..

Comment (CPerezz) [reply zkevm-circuits/src/witness.rs]: I think if you make that PR and split the stuff you need to also add anything that is required TBH.

Comment (lispc) [reply zkevm-circuits/src/witness.rs]: No there was no docs at least i did not make things worse. Tbh at least now i have no plan to add docs for rw

**Comment by CPerezz** [comment]
@lispc if you can resolve the conflicts, let's merge this.

**Pull request merged**